src/avatar/internal.tsx

```diff
@@ -0,0 +1,117 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+import React, { useState, useRef } from 'react';
+import clsx from 'clsx';
+import { warnOnce } from '@cloudscape-design/component-toolkit/internal';
+
+import { getBaseProps } from '../internal/base-component/index.js';
+import { InternalBaseComponentProps } from '../internal/hooks/use-base-component';
+import Tooltip from '../internal/components/tooltip';
+
+import LoadingDots from './loading-dots';
+import InternalIcon from '../icon/internal.js';
+import { AvatarProps } from './interfaces.js';
+import styles from './styles.css.js';
+
+export interface InternalAvatarProps extends AvatarProps, InternalBaseComponentProps {}
+
+const AvatarContent = ({ color, loading, initials, iconName, iconSvg, iconUrl, ariaLabel }: AvatarProps) => {
+  if (loading) {
+    // TODO: check spinner tests
+    return <LoadingDots color={color} />;
+  }
+
+  if (initials) {
+    const letters = initials.length > 2 ? initials.slice(0, 2) : initials;
+
+    if (initials.length > 2) {
+      warnOnce('Avatar', `"initials" is longer than 2 characters. Only the first two characters are shown.`);
+    }
+
+    return <span>{letters}</span>;
+  }
+
+  return (
+    <div>
+      <InternalIcon name={iconName || 'user-profile'} svg={iconSvg} url={iconUrl} alt={ariaLabel} />
+    </div>
+  );
+};
+
+export default function InternalAvatar({
+  color,
+  tooltipText,
+  initials,
+  loading = false,
+  ariaLabel,
+  iconName,
+  iconSvg,
+  iconUrl,
+  __internalRootRef = null,
+  ...rest
+}: InternalAvatarProps) {
+  const baseProps = getBaseProps(rest);
+  const handleRef = useRef<HTMLDivElement>(null);
+  const [showTooltip, _setShowTooltip] = useState(false);
+
+  const setShowTooltip = (value: boolean) => {
+    if (tooltipText) {
+      return _setShowTooltip(value);
```

Comment:
Alternatively, we can keep the default setShowTooltip but only render it when `showTooltip && tooltipText`

Reply:
Indeed. Updated.